uniffi_bindgen/src/bindings/swift/templates/RustBufferHelper.swift

```diff
@@ -225,6 +225,57 @@ extension Bool: ViaFfi {
     }
 }
 
+extension Date: ViaFfiUsingByteBuffer, ViaFfi {
+    fileprivate static func read(from buf: Reader) throws -> Self {
+        let seconds: UInt64 = try buf.readInt()
+        let microseconds: UInt32 = try buf.readInt()
+        let delta = Double(seconds) + (Double(microseconds) / 1.0e9)
+        return Date.init(timeIntervalSince1970: delta)
+    }
+
+    fileprivate func write(into buf: Writer) {
+        let delta = self.timeIntervalSince1970
+        if (delta < 0) {
+            fatalError("Invalid timestamp, must be greater than UNIX EPOCH")
+        }
+        let seconds = UInt64(delta)
+        let microseconds = UInt32((delta - Double(seconds)) * 1.0e9)
+        buf.writeInt(seconds)
+        buf.writeInt(microseconds)
+    }
+}
+
+extension TimeInterval {
+    fileprivate static func liftTimeInterval(_ buf: RustBuffer) throws -> Self {
```

Comment:
You may need to change the name of this method to match suggested use of `canonical_type_name` above.